File: sdks/java/basys.sdk/src/org/eclipse/basyx/aas/metamodel/hashmap/aas/qualifier/Identifiable.java
package org.eclipse.basyx.aas.metamodel.hashmap.aas.qualifier;

import org.eclipse.basyx.aas.api.metamodel.aas.identifier.IIdentifier;
import org.eclipse.basyx.aas.api.metamodel.aas.qualifier.IAdministrativeInformation;
import org.eclipse.basyx.aas.api.metamodel.aas.qualifier.IIdentifiable;
import org.eclipse.basyx.aas.metamodel.facades.IdentifiableFacade;
import org.eclipse.basyx.aas.metamodel.hashmap.aas.identifier.Identifier;

/**
 * Identifiable class
 * 
 * @author kuhn, schnicke
 *
 */
public class Identifiable extends Referable implements IIdentifiable  {

	/**
	 * Version of serialized instances
	 */
	private static final long serialVersionUID = 1L;
	
	public static final String ADMINISTRATION="administration";
	
	public static final String IDENTIFICATION="identification";

	/**
	 * Default constructor
	 */
	public Identifiable() {
		// Add qualifier
		putAll(new Referable());

		// Administrative information of an element. (AdministrativeInformation)
		put(ADMINISTRATION, new AdministrativeInformation());
		// The globally unique identification of an element. (Identificator)
		put(IDENTIFICATION, new Identifier());
	}

	/**
	 * Constructor that accepts values for most relevant properties
	 */
	public Identifiable(String version, String revision, String idShort, String category, String description, String idType, String id) {
		// Add qualifier
		putAll(new Referable(idShort, category, description));

		// Create administrative information of an element. (AdministrativeInformation)
		put(ADMINISTRATION, new AdministrativeInformation(version, revision));
		// The globally unique identification of an element. (Identificator)
		put(IDENTIFICATION, new Identifier(idType, id));
	}

	@Override
	public IAdministrativeInformation getAdministration() {
	return new IdentifiableFacade(this).getAdministration();
	}

	@Override
	public IIdentifier getIdentification() {
		return new IdentifiableFacade(this).getIdentification();
	}

	@Override
	public void setAdministration(String version, String revision) {
		new IdentifiableFacade(this).setAdministration(version, revision);
		
	}

	@Override
	public void setIdentification(String idType, String id) {
		new IdentifiableFacade(this).setIdentification(idType, id);
		
	}


}

File: sdks/c++/basys.sdk.cc/src/aas/impl/metamodel/hashmap/aas/qualifier/Identifiable.cpp
/*
 * Identifiable.cpp
 *
 *      Author: wendel
 */

#include "Identifiable.h"

#include "AdministrativeInformation.h"
#include "impl/metamodel/hashmap/aas/identifier/Identifier.h"

namespace basyx {
namespace submodel {
namespace metamodel {
namespace map {
namespace qualifier {

using namespace aas::qualifier;
using namespace aas::identifier;

Identifiable::Identifiable() :
  administration {new aas::qualifier::impl::AdministrativeInformation},
  identification {new aas::identifier::impl::Identifier}
{}

Identifiable::Identifiable(const std::string & version, const std::string & revision, const std::string & idShort, 
    const std::string & category, const aas::qualifier::impl::Description & description, const std::string & idType, const std::string & id) :
  Referable {idShort, category, description},
  administration {new aas::qualifier::impl::AdministrativeInformation {version, revision} },
  identification {new aas::identifier::impl::Identifier {id, idType} }
{}

std::shared_ptr<IAdministrativeInformation> Identifiable::getAdministration() const
{
  return this->administration;
}

std::shared_ptr<IIdentifier> Identifiable::getIdentification() const
{
  return this->identification;
}

void Identifiable::setAdministration(const std::shared_ptr<aas::qualifier::IAdministrativeInformation> & administration)
{
  this->administration = administration;
}

void Identifiable::setIdentification(const std::shared_ptr<aas::identifier::IIdentifier> & identification)
{
  this->identification = identification;
}

}
}
}
}
}
